notebooks/00_quick_start/sar_aml_compute_movielens.ipynb

```diff
@@ -0,0 +1,477 @@
+    "    \"col_user\": \"UserId\",\n",
+    "    \"col_item\": \"MovieId\",\n",
+    "    \"col_rating\": \"Rating\",\n",
+    "    \"col_timestamp\": \"Timestamp\",\n",
+    "}\n",
+    "\n",
+    "logging.basicConfig(level=logging.DEBUG, \n",
+    "                    format='%(asctime)s %(levelname)-8s %(message)s')\n",
+    "\n",
+    "model = SARSingleNode(\n",
+    "    remove_seen=True, similarity_type=\"jaccard\", \n",
+    "    time_decay_coefficient=30, time_now=None, timedecay_formula=True, **header\n",
+    ")\n",
+    "\n",
+    "# train the SAR model\n",
+    "start_time = time.time()\n",
+    "\n",
+    "model.fit(train)\n",
+    "\n",
+    "train_time = time.time() - start_time\n",
+    "run.log(name=\"Training time\", value=\"Took {} seconds for training.\".format(train_time))\n",
+    "\n",
+    "start_time = time.time()\n",
+    "\n",
+    "top_k = model.recommend_k_items(test)\n",
+    "\n",
+    "test_time = time.time() - start_time\n",
+    "run.log(name=\"Prediction time\", value=\"Took {} seconds for prediction.\".format(test_time))\n",
+    "\n",
+    "# TODO: remove this call when the model returns same type as input\n",
+    "top_k['UserId'] = pd.to_numeric(top_k['UserId'])\n",
+    "top_k['MovieId'] = pd.to_numeric(top_k['MovieId'])\n",
+    "\n",
+    "# evaluate\n",
+    "eval_map = map_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                    col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                    relevancy_method=\"top_k\", k=args.top_k)\n",
+    "eval_ndcg = ndcg_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                      col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                      relevancy_method=\"top_k\", k=args.top_k)\n",
+    "eval_precision = precision_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                                col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                                relevancy_method=\"top_k\", k=args.top_k)\n",
+    "eval_recall = recall_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                          col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                          relevancy_method=\"top_k\", k=args.top_k)\n",
+    "\n",
+    "run.log(name=\"Model\", value=model.model_str)\n",
+    "run.log(name=\"Top K\", value=args.top_k)\n",
+    "run.log(name=\"MAP\", value=eval_map)\n",
+    "run.log(name=\"NDCG\", value=eval_ndcg)\n",
+    "run.log(name=\"Precision@K\", value=eval_precision)\n",
+    "run.log(name=\"Recall@K\", value=eval_recall)\n",
+    "\n",
+    "pm.record(\"map\", eval_map)\n",
+    "pm.record(\"ndcg\", eval_ndcg)\n",
+    "pm.record(\"precision\", eval_precision)\n",
+    "pm.record(\"recall\", eval_recall)\n",
+    "pm.record(\"train_time\", train_time)\n",
+    "pm.record(\"test_time\", test_time)\n",
+    "\n",
```

Comment:
Do we want to show both run.log and pm.record? It seems very long-whinded

Reply:
I used run.log() the values to see in Experiment. Now deleted run.log